### ISSUE TITLE
Print dirty bytes for removed variables

### ISSUE THREAD

**Issue by ZeroEkkusu** [0xPolygon/storage-layout-checker#11]
If a variable is removed (not overridden), print dirty bytes ~~to mask~~.

## Context

Let's say we start with:

```
    bytes1 a
    bytes2 b
```

<br>

**If we remove `b`:**

```
    bytes1 a
🪦  2 dirty bytes
```

_Case: `a` doesn't overlap `b`_

In this case we print how many dirty bytes there are.

<br>

**If we change `a` to `bytes2`, and remove `b`:**

```
🏴  bytes2 a
🪦
    1 dirty byte
```

_Case: `a` partially overlaps `b`_

In this case we don't print how many dirty bytes there are in the overlap, only after it.

<br>

**If we change `a` to `bytes3`, and remove `b`:**

```
🏴  bytes3 a
🪦
```

_Case: `a` completely overlaps `b`_

In this case we don't print how many dirty bytes there are.

<br>

I can suggest one way to solve this:

Create a new function `addDirtyAfterRemoved` that takes two inputs: the old and new layouts.

By that, I mean these two variables here:

https://github.com/0xPolygon/storage-delta/blob/982d94742c0595d48efb04e083e019bf701370c0/_reporter.js#L30

Call the function immediately after the above line.

What it should do:

For each item in the new layout, check if it's undefined.
If it is, add a dirty item after it to the new layout and an undefined item to the old layout, if applicable, according to the rules we set above.

That should take care of the new dirty items.
For printing dirty bytes at the same line as 🪦, I think we'll have to handle that on the 'ANALYZE' level instead; I mean here:

https://github.com/0xPolygon/storage-delta/blob/982d94742c0595d48efb04e083e019bf701370c0/_reporter.js#L41

For example by adding a helper function that checks whether we should just print 🪦 to the report with an empty line, or with the dirty bytes info.

<br>

I know this can get complex at times, so feel free to ping me to help you out.

## Additional notes

Beware of a situation like this one (I'll add positions for clarity):

```
0     bytes2 a
2     bytes4 b
```

Change to:

```
0     bytes3 a
3     bytes1 c
```

In this case, the finding should be:

```
0 🏴  bytes3 a
2 🪦
3 🏴  bytes1 c
4     2 dirty bytes
```

**Comment by triggeredcode** [comment]
@ZeroEkkusu Can you assign this to me? Also I am a beginner so some context would be nice. Thanks

**Comment by ZeroEkkusu** [comment]
Gm, @triggeredcode.

Yes, I can! Though, you may find #7 more interesting (and it's more useful).

I'll add some description there, in case you change your mind. If not, let me know, and we'll return to this one!

**Comment by kamuik16** [comment]
> If a variable is removed (not overridden), print dirty bytes to mask.

Hey @ZeroEkkusu ! Is this issue still open for contribution? I would like to work on this.

**Comment by ZeroEkkusu** [comment]
@kamuik16 Replied to you in #9!

**Comment by HarishPrasannaV** [comment]
Hey, I'm a beginner and I'd like to contribute too.

**Comment by kamuik16** [comment]
> @kamuik16 Replied to you in #9!

Ready to do this, some context and what output is needed would be great!

**Comment by ZeroEkkusu** [comment]
@kamuik16 Added!

**Comment by ZeroEkkusu** [comment]
@kamuik16, just a heads up that I think I've spotted an edge case. I'll modify the description to explain it. Also, I'll remove the second part (legacy) for now, if that's okay with you.

**Comment by kamuik16** [comment]
> @kamuik16, just a heads up that I think I've spotted an edge case. I'll modify the description to explain it. Also, I'll remove the second part (legacy) for now, if that's okay with you.

Yeah sure.

**Comment by ZeroEkkusu** [comment]
@kamuik16 Explained.

Please feel free to ping me if I can help with the development.

**Comment by kamuik16** [comment]
> @kamuik16 Explained.
> 
> Please feel free to ping me if I can help with the development.

Okay.

**Comment by ZeroEkkusu** [comment]
Btw, if this is complicated, https://github.com/0xPolygon/storage-delta/issues/8 may be another good choice.

I've came up with an easier way to solve it and will describe it there as well.

**Comment by kamuik16** [comment]
> Btw, if this is complicated, #8 may be another good choice.
> 
> I've came up with an easier way to solve it and will describe it there as well.

Yeah, I guess this is complicated for me as I am yet to take a deeper understanding of the whole script.

**Comment by ZeroEkkusu** [comment]
No worries, that's totally understandable.
